runtime_scan/pkg/provider/gcp/config.go

```diff
@@ -0,0 +1,69 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/viper"
+)
+
+const (
+	DefaultEnvPrefix = "VMCLARITY_GCP"
+)
+
+type Config struct {
+	ProjectID         string `mapstructure:"project_id"`
+	ScannerZone       string `mapstructure:"scanner_zone"`
+	ScannerSubnetwork string `mapstructure:"scanner_subnetwork"`
+}
+
+func NewConfig() (Config, error) {
+	// Avoid modifying the global instance
+	v := viper.New()
+
+	v.SetEnvPrefix(DefaultEnvPrefix)
+	v.AllowEmptyEnv(true)
+	v.AutomaticEnv()
+
+	_ = v.BindEnv("project_id")
+	_ = v.BindEnv("scanner_zone")
+	_ = v.BindEnv("scanner_subnetwork")
+
+	config := Config{}
+	if err := v.Unmarshal(&config, viper.DecodeHook(mapstructure.TextUnmarshallerHookFunc())); err != nil {
+		return Config{}, fmt.Errorf("failed to parse provider configuration. Provider=GCP: %w", err)
+	}
+	return config, nil
+}
+
+// nolint:cyclop
+func (c Config) Validate() error {
+	if c.ProjectID == "" {
+		return fmt.Errorf("parameter ProjectID must be provided via env var VMCLARITY_GCP_PROJECT_ID")
+	}
+
+	if c.ScannerZone == "" {
+		return fmt.Errorf("parameter ScannerZone must be provided via env var VMCLARITY_GCP_SCANNER_VM_SIZE")
+	}
+
+	if c.ScannerSubnetwork == "" {
+		return fmt.Errorf("parameter ScannerSubnetwork must be provided via env var VMCLARITY_GCP_SCANNER_SUBNETWORK")
```

Comment:
where is `VMCLARITY_GCP_PROJECT_ID`, `VMCLARITY_GCP_SCANNER_VM_SIZE` and `VMCLARITY_GCP_SCANNER_SUBNETWORK` defined?

Also, `VMCLARITY_GCP_SCANNER_VM_SIZE` is not `VMCLARITY_GCP_SCANNER_ZONE`?

Reply:
> @FrimIdan Env vars defined here: https://github.com/openclarity/vmclarity/pull/433/files#diff-db8032f507629e5a7bb5e3459ed38e1ae146cb3e7dd2f56324465ab402407b06R39-R45

@chrisgacsal Problem that it's not a const so can't use the name here you then when changing the env var the print here will be wrong.
For example even now `VMCLARITY_GCP_SCANNER_VM_SIZE` is wrong I think, it should be `VMCLARITY_GCP_SCANNER_ZONE`, right?

Reply:
Will add consts for these

Reply:
@FrimIdan switched to use consts

Reply:
@fishkerez sure, they just all need to be aligned with the input to `BindEnv` and `SetEnvPrefix`
I mean maybe something like
```
const (
	DefaultEnvPrefix = "VMCLARITY_GCP"
	projectID = "project_id"
)

var projectIDEnvVar = strings.Join([]string{DefaultEnvPrefix, strings.ToUpper(projectID)}, "_")
.
.
.
	v.SetEnvPrefix(DefaultEnvPrefix)=

	_ = v.BindEnv(projectID)
```

Reply:
Done